### PR TITLE
Update Router.php

### PR DESCRIPTION
**Pull request by lai0n**

### CODE REVIEW

**Comment by bramus** [comment]
Hmmz, tests seem to be failing:

```
$ phpunit
PHPUnit 4.8.18 by Sebastian Bergmann and contributors.
.F............R.....E..

Time: 550 ms, Memory: 23.25Mb

There was 1 error:

1) RouterTest::testSubfolders
Cannot modify header information - headers already sent by (output started at phar:///home/travis/.phpenv/versions/5.3.29/bin/phpunit/phpunit/Util/Printer.php:134)
/home/travis/build/bramus/router/src/Bramus/Router/Router.php:261
/home/travis/build/bramus/router/tests/RouterTest.php:566

--

There was 1 failure:

1) RouterTest::testUri
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'/about/whatever'
+'/sub/folder/about/whatever'
/home/travis/build/bramus/router/tests/RouterTest.php:49

FAILURES!

Tests: 23, Assertions: 44, Errors: 1, Failures: 1, Risky: 1.

The command "phpunit" exited with 2.

```

**Comment by sant0ro** [comment]
What this Pull Request really changes? @lai0n

File: src/Bramus/Router/Router.php [hunk 1]
```diff
@@ -362,7 +362,9 @@ private function handle($routes, $quitAfterRun = false)
     protected function getCurrentUri()
     {
         // Get the current Request URI and remove rewrite base path from it (= allows one to run the router in a sub folder)
-        $uri = substr($_SERVER['REQUEST_URI'], strlen($this->getBasePath()));
+        $uri =  $uri = urldecode(
```
Comment (sant0ro) [inline src/Bramus/Router/Router.php]: Why `$uri = $uri` ?

**Comment by bramus** [comment]
@sant0ro Main change is that it uses the predefined `PHP_URL_PATH` constant, instead of calculating the base path itself. It _should_ fix some problems with subfolders (see #30), yet the test don't seem to back this.

**Comment by sant0ro** [comment]
@bramus i see...

**Comment by sant0ro** [comment]
Weird i can see what is wrong. Why the PHP headers was already sent?

**Comment by sant0ro** [comment]
One of the tests resulted in failure, 
<pre>
        $_SERVER['SCRIPT_NAME'] = '/sub/folder/index.php';
        $_SERVER['REQUEST_URI'] = '/sub/folder/about/whatever';
</pre>
Are those lines correct?

**Comment by lai0n** [comment]
The tests are incorrect, to fix:
`       $_SERVER['SCRIPT_NAME'] = '/sub/folder/index.php';
        $_SERVER['REQUEST_URI'] = '/about/whatever';`

**Comment by sant0ro** [comment]
@lai0n you need be totally sure with those variables. I recommend simulating it on a real scenario, or reading the PHP Documentation.

**Comment by lai0n** [comment]
@sant0ro In a real scenario, the code works fine in subfolder or not.

**Comment by bramus** [comment]
The tests indeed fake some of the values in the `$_SERVER` superglobal. Could be possible that they are not set correctly to allow `PHP_URL_PATH` to work properly. This needs some investigation.

**Pull request closed without merging**